static/js/shared/data_fetcher.ts

```diff
@@ -387,108 +146,112 @@ function computePerCapita(
  * variables.
  *
  * @param places A list of place dcids.
- * @param statsVars A list of statistical variable dcids.
+ * @param statVars A list of statistical variable dcids.
  * @param perCapita Whether to compute per capita result.
  * @param scaling Scaling factor for per capita compuation.
  *
- * @returns A Promise of StatsData object.
+ * @returns A Promise of StatData object.
  */
-function fetchStatsData(
+export function fetchStatData(
   places: string[],
-  statsVars: string[],
+  statVars: string[],
   perCapita = false,
   scaling = 1,
-  denominators: string[] = [],
-  cachedData: CachedStatVarDataMap = {}
-): Promise<StatsData> {
-  if (denominators.length && denominators.length != statsVars.length) {
-    console.log(
-      "StatVars must have the same number of denominators, if specified"
-    );
-    return;
+  denominators: Record<string, string> = {}
+): Promise<StatData> {
+  let denomStatVars = [];
+  if (perCapita) {
+    denomStatVars.push(TOTAL_POPULATION_SV);
+  } else {
+    for (const sv in denominators) {
+      denomStatVars.push(denominators[sv]);
+    }
   }
+  denomStatVars = Array.from(new Set(denomStatVars));
 
-  if (isAllCachedDataAvailable(places, statsVars, denominators, cachedData)) {
-    return new Promise((resolve) => {
-      resolve(
-        getStatsDataFromCachedData(
-          places,
-          statsVars,
-          perCapita,
-          scaling,
-          denominators,
-          cachedData
-        )
-      );
+  const statDataPromise: Promise<StatApiResponse> = axios
+    .post(`/api/stats`, {
+      places: places,
+      statVars: statVars,
+    })
+    .then((resp) => {
+      return resp.data;
     });
+  let denomDataPromise: Promise<StatApiResponse>;
+  if (denomStatVars.length > 0) {
+    denomDataPromise = axios
+      .post(`/api/stats`, {
+        places: places,
+        statVars: denomStatVars,
```

Comment:
should we combine this in the previous call to /api/stats to reduce the number of requests? or a todo, for a followup if there are performance issues

Reply:
I tried to do that but that makes the code a bit hard to understand. Need to split the response based into denominator and numerator groups and they could have overlap. This is not too bad since this is adding at most one API call (only happens for per capita) and executed in parallel.